projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_dag.py

```diff
@@ -134,3 +136,110 @@ def _start_job(self, node):
                 self._delayed_starting_jobs.enqueue(node)
             else:
                 raise
+
+    def _validate_job(self, job: Dict):
+        allowed_job_keys = [
+            "job_name",
+            "team_name",
+            "fail_meta_job_on_error",
+            "depends_on",
+        ]
+        required_job_keys = ["job_name", "depends_on"]
+        registered_job_names = [j.job_name for j in self._job_executor.get_all_jobs()]
+        if type(job) != dict:
+            raise UserCodeError(
+                ErrorMessage(
+                    "",
+                    "Meta Job failed due to a data job validation failure.",
+                    f"Validation of Data Job failed. "
+                    f"There is an error with the type of job name.",
+                    "The DAG will not be built and the meta data job will fail.",
+                    f"Check the Data Job type. Expected type is dict.",
+                )
+            )
+        if any(key not in allowed_job_keys for key in job.keys()) or any(
+            key not in job for key in required_job_keys
+        ):
+            raise UserCodeError(
+                ErrorMessage(
+                    "",
+                    "Meta Job failed due to a data job validation failure.",
+                    f"Validation of Data Job failed. "
+                    "Some keys in the Data Job Dict are not allowed or some of the required ones are missing.",
+                    "The DAG will not be built and the Meta Job will fail.",
+                    "Check the Data Job Dict keys for errors and make sure the required ones are present.",
```

Comment:
It would be good to list which keys are allowed and which are required to make debugging easier.

Reply:
Makes sense, divided them into two conditions. Will push soon.